src/modules/vault/hooks/assets/useVaultAssets.ts

```diff
@@ -1,66 +1,41 @@
 import { Vault } from 'bakosafe';
 import { bn } from 'fuels';
-import { useCallback, useEffect, useMemo } from 'react';
+import { useCallback, useMemo, useState } from 'react';
 import { useQuery } from 'react-query';
 
-import BakoIcon from '@/assets/tokens/bako.svg';
 import { useAuth } from '@/modules/auth/hooks';
-import { Asset, assetsMap, ETHDefault, NativeAssetId } from '@/modules/core';
+import { assetsMap, ETHDefault, NativeAssetId } from '@/modules/core';
+
+const IS_VISIBLE_KEY = '@bakosafe/balance-is-visible';
+
+const isVisibleBalance = () => localStorage.getItem(IS_VISIBLE_KEY) === 'true';
+const setIsVisibleBalance = (isVisible: 'true' | 'false') =>
+  localStorage.setItem(IS_VISIBLE_KEY, isVisible);
 
 import { VaultService } from '../../services';
-import { useVaultState } from '../../states';
 
-const balancesToAssets = async (
-  setBalanceUSD: (string: string) => void,
-  predicate?: Vault,
-) => {
-  if (!predicate) return [];
+const balancesToAssets = async (predicate?: Vault) => {
+  if (!predicate) return {};
 
-  const balances = await predicate.getBalances();
-  const { reservedCoins, balanceUSD } = await VaultService.hasReservedCoins(
-    predicate.BakoSafeVaultId,
-  );
-  setBalanceUSD(balanceUSD);
-  const result = balances.reduce((acc, balance) => {
-    const assetInfos = assetsMap[balance.assetId];
-    const reservedCoinAmount = reservedCoins?.find(
-      (item) => item.assetId === balance.assetId,
-    )?.amount;
-    const adjustedAmount = reservedCoinAmount
-      ? balance.amount.sub(reservedCoinAmount)
-      : balance.amount;
-
-    if (adjustedAmount.gt(0)) {
-      acc.push({
-        amount: adjustedAmount.format(),
-        slug: assetInfos?.slug ?? 'UKN',
-        name: assetInfos?.name ?? 'Unknown',
-        assetId: balance.assetId,
-        icon: assetInfos?.icon ?? BakoIcon,
-      });
-    }
-
-    return acc;
-  }, [] as Required<Asset>[]);
-
-  return result;
+  const { currentBalanceUSD, currentBalance } =
```

Comment:
use this info by context...if this part dont used, remove

Reply:
it's used in the `useVaultAssets` function and in many places in vaults page. Balance Cards, CardDetails in the vault home, for example.
And yes, it is being used by context